### DISCUSSION TITLE
Add and show a CSV data file on Quarto website

### DISCUSSION THREAD

**Discussion by javedali99** [quarto-dev/quarto-cli#1533] [Q&A]
Hi. I want to upload a CSV file that includes a few hundreds of data entries (rows) on my quarto website in a way it can be fully shown on the website and also add a search/query option for that CSV dataset.

Can anyone help me with how to do it? I am new to quarto. 

Thank you.

**Comment by rgaiacs** [comment]
Hi @javedali99,

to add a search/query option you will have to include some Javascript like https://gridjs.io/ and http://tabulator.info/.

**Comment by dragonstyle** [comment]
As an alternative proposal, you could consider converting the CSV to yaml and using a `listing` to display that:

https://quarto.org/docs/websites/website-listings-custom.html#metadata-file-listings

The 'table' style listing is designed to allow any number of fields and to handle sorting and filtering by them (though you will need to provide the list of fields you'd like displayed and the data-types of the fields if there is special ordering/filtering treatment that you'd like them receive).

**Comment by javedali99** [comment]
Thank you very much @dragonstyle @rgaiacs. I will try these methods and see which one is more suitable for me.
